app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -125,8 +125,17 @@ def show
 
     api :POST, "/repositories/:id/sync", N_("Sync a repository")
     param :id, :identifier, :required => true, :desc => N_("repository ID")
+    param :source_url, String, :desc => N_("temporarily override feed URL for sync"), :required => false
     def sync
-      task = async_task(::Actions::Katello::Repository::Sync, @repository)
+      if params[:source_url].present? && params[:source_url] !~ /\A#{URI.regexp}\z/
+        fail HttpErrors::BadRequest, N_("source URL is malformed")
+      end
+
+      if params[:source_url].blank? && @repository.url.blank?
+        fail HttpErrors::BadRequest, N_("attempted to sync without any feed URL")
```

Comment:
Another option is to simply rescue within this function and raise this error code, e.g.:

```
def sync

rescue => e
  raise HttpErrors::BadRequest e.message
end
```